### DISCUSSION TITLE
multilevel list in tailwind

### DISCUSSION THREAD

**Discussion by pavelhynek** [tailwindlabs/tailwindcss#7476] [Help] [answered]
Hello,

I can't think of a better way to create a multilevel list in tailwind for a while now.

The problem arises when he made a multi-level list that all levels have no indentation and are left-aligned. What I don't understand is whether to set each `<ul>` margin to achieve the native behavior of list. My goal is to create a multi-level according to the UX/UI design below.

![list](https://user-images.githubusercontent.com/17236984/154061431-43687d92-cfab-4f10-9482-33dab1414e11.PNG)

If I make a list with three levels it looks like this:

![list 2](https://user-images.githubusercontent.com/17236984/154223230-6e0ad374-90c5-49b8-b1e5-6c33847daf1e.PNG)

```   
    <ul class="list-decimal list-inside">
      <li>Now this is a story all about how, my life got flipped turned upside down</li>
      <ul class="list-alfa list-inside">
        <li>abeceda</li>
        <ul class="list-roman list-inside">
          <li>římské</li>
        </ul>
      </ul>
      <li>I'll tell you how I became the prince of a town called Bel-Air </li>
    </ul>

```  
I am new in tailwind css and appreciate any help thanks  ❤.

**Comment by rflw** [comment]
Please, add more details, what the problem is and what is the goal?

**Reply by pavelhynek** [reply]
Ok thanks I add more info.

**Comment by imprisonedmind** [comment, marked as answer]
You Can use ml or pl to assign margins to the left to achieve indentation. 

![image](https://user-images.githubusercontent.com/51891628/154231067-c831c7cf-7a14-4a79-939f-7591fa3ce2a2.png)
```
  <ul class="list-decimal list-inside">
      <li>Parent using margins</li>
      <ul class="list-disc list-inside ml-4">
        <li>Child 0</li>
        <li>Child 1</li>
        <ul class="list-disc list-inside ml-8">
          <li>Child of Child</li>
          <li>Child of Child</li>
          <li>Child of Child</li>
        </ul>
        <li>Child 2</li>
        <ul class="list-disc list-inside ml-8">
          <li>Child of Child</li>
        </ul>
      </ul>
      <li>Second Parent Using Padding</li>
      <ul class="list-disc list-inside pl-4">
        <li>Child 0</li>
        <li>Child 1</li>
        <ul class="list-disc list-inside pl-8">
          <li>Child of Child</li>
          <li>Child of Child</li>
          <li>Child of Child</li>
        </ul>
        <li>Child 2</li>
        <ul class="list-disc list-inside pl-8">
          <li>Child of Child</li>
        </ul>
      </ul>
    </ul>
```

**Reply by MatheusRoichman** [reply]
hey, same problem here. but I can't do it like that, because the list is dynamic (comes from CMS). what styles do I need to set in order for it to appear multi-leveled normally?
I tested on a blank HTML page and it is the default behavior.

**Reply by imprisonedmind** [reply]
After doing some testing, it may be as simple as this:

```scss
ul {
  @apply pl-4 list-disc
}
```

View the tailwind playground [here](https://play.tailwindcss.com/RteK1TPOtD)

<img width="133" alt="Screenshot 2024-06-01 at 02 12 36" src="https://github.com/tailwindlabs/tailwindcss/assets/51891628/a666f33e-d6b5-444d-87a3-278b024eec2b">

**Reply by MatheusRoichman** [reply]
ah yes, it worked. tks a lot @imprisonedmind :)
